website/source/docs/internals/providers/new-resource.html.md

```diff
@@ -0,0 +1,127 @@
+easy to find.
+
+## Registering the Resource With the Provider
+
+Now that you've got your function, switch back to your Provider definition, and
+add the Resource to the Provider's `ResourcesMap` property. The key will be the
+name of the Resource used in state and configuration files, and (by convention)
+should match the `{resource_name}` part used in the filename for the file that
+defines the Resource. Nothing will necessarily break if you don't do this, but
+if you happen to accidentally have a collision with any other resource in any
+other provider, things will likely break. This convention helps keep things
+neat and orderly while avoiding conflicts.
+
+## Defining the Resource Properties
+
+Each Resource has a set of properties called the "schema" that is stored in the
+state. Think of it as the type definition for the Resource. For example, an AWS
+instance resource has a `name` property, to control the name of the instance,
+GitHub's repository resource has a `default_branch` property to set the default
+branch for the repository, and so on.
+
+These properties are defined in the `Schema` property of the Resource
+definition. It takes a map with the keys being the property name and the values
+being `*Schema` structs. The `*Schema` structs define some type information
+(what kind of data to expect, etc.) along with some [advanced
+behaviour](/docs/internals/providers/schema.html) for resources that helps
+Terraform do the right thing without you needing to write a bunch of code.
+
+~> **Note:** "id" is a reserved property name. Don't call your property "id".
+
+## Calling the API
+
+Now that we have a Resource fully defined, it's time to make it do something.
+We're going to use the Provider's API client (possibly
+[configured](/docs/internals/providers/new-provider.html#configuring-your-provider)
+by the Provider's `ConfigureFunc`) to create, read, update, destroy, and check
+for the existence of some resources. We do this by defining functions to
+perform those operations for us. Terraform will then decide which of the
+functions to call and on which resources. All the functions have to do is know
+how to determine and set the state of a resource using the Provider's API.
+
+Each function takes a `*ResourceData` struct and an `interface{}` as arguments,
+and returns an `error`. The only exception is `ExistsFunc`, which takes the
+same arguments, but returns a boolean or an `error`.
+
+The `*ResourceData` struct contains the state of the Resource as it should be.
+It's an amalgamation of several different sources of data, which are explained
+further in [Using ResourceData](/docs/internals/providers/resource-data.html).
+
+The `interface{}` is the same `interface{}` returned by the Provider's
+`ConfigureFunc`. Typically, this is where you'd put the configured API client,
+for example.
+
+Of all these functions, only the `Update` and `Exists` functions are optional.
```

Comment:
There are definitely other functions on the helper/schema - `Importer`, `Timeouts`, `MigrateState`